### PR TITLE
test_target_teams_thread_limit.c: Fix out of bounds

### PR DESCRIPTION
**Pull request by tob2**
Two simple fixes:
* Use OMPVV_NUM_TEAMS_DEVICE for the 'errors' array as it is iterated over teams. Before the patch, it has size 1 which is too small.
* Fix a typo

* * *

Found via GCC's warning `test_target_teams_thread_limit.c:36:25: warning: iteration 1 invokes undefined behavior`.

The problem is that `OMPVV_NUM_THREADS_DEVICE == 8` and `OMPVV_NUM_TEAMS_DEVICE == 8` such that the array has size 1 at
https://github.com/SOLLVE/sollve_vv/blob/6a7dcb7217aea5db5bbd5d8c63ccafa7a6544244/tests/5.1/teams/test_target_teams_thread_limit.c#L23

And, thus, the following two loops access the array out of bounds:

https://github.com/SOLLVE/sollve_vv/blob/6a7dcb7217aea5db5bbd5d8c63ccafa7a6544244/tests/5.1/teams/test_target_teams_thread_limit.c#L35-L36
https://github.com/SOLLVE/sollve_vv/blob/6a7dcb7217aea5db5bbd5d8c63ccafa7a6544244/tests/5.1/teams/test_target_teams_thread_limit.c#L59-L60

@nolanbaker31 @spophale @jrreap @krishols @mjcarr458 @seyonglee – please review

### CODE REVIEW

**Review by nolanbaker31 — APPROVED** [review]
lgtm

**Comment by tob2** [comment]
**ping**
@spophale @jrreap @krishols @mjcarr458 @seyonglee – please review; Nolan already had the first "lgtm", but a second review is required.

**Comment by seyonglee** [comment]
These changes look good to me too.
However, the comments at the top seem to be wrong; they say that "This test uses the thread_limit clause on the teams construct", but `thread_limit` clause is applied to the `target` construct, instead of `teams` construct.

**Comment by tob2** [comment]
> comments at the top seem to be wrong; they say that "This test uses the thread_limit clause on the teams construct",

I have now corrected this preexisting copy'n'paste error.

File: tests/5.1/teams/test_target_teams_thread_limit.c
Comment (seyonglee) [inline tests/5.1/teams/test_target_teams_thread_limit.c:6]: Delete "above" from "a above target construct"?

**Review by seyonglee — APPROVED** [review]
It looks good to me.

**Comment by tob2** [comment]
@spophale – PING – this has two approvals but is not yet merged.

**Comment by tob2** [comment]
@spophale – **PING²** – this has two approvals but is not yet merged.

**Pull request merged**
